package/MDAnalysis/core/AtomGroup.py

```diff
@@ -2825,11 +2825,199 @@ def wrap(self, compound="atoms", center="com", box=None):
                 o.translate(s)
 
     def select_atoms(self, selstr, *othersel, **selgroups):
-        """Selection of atoms using the MDAnalysis selection syntax.
+        """Select atoms using a CHARMM selection string.
 
-        AtomGroup.select_atoms(selection[,selection[,...]], [groupname=atomgroup[,groupname=atomgroup[,...]]])
+        Parameters
+        ----------
+        sel : str
+            The selection string to be parsed (following the language below)
+        *othersel : str, optional
+            More selection strings to be parsed
+        **selgroups : AtomGroup(s), optional
+            groups to be referred to using the 'group' keyword
 
-        .. SeeAlso:: :meth:`Universe.select_atoms`
+        Returns
+        -------
+        A :class:`AtomGroup`, with atoms sorted according to their
+        index.  This AtomGroup will not contain any duplicate atoms,
+        even if the original selection had so.
+
+        Note
+        ----
+        If exact ordering of atoms is required (for instance, for
+        :meth:`~AtomGroup.angle` or :meth:`~AtomGroup.dihedral` calculations)
+        then one supplies selections *separately* in the required order.
+        For example::
+           ag.select_atoms('name HT*', 'name N')
+        Gives anything matching HT* before anything matching N
+
+        Examples
+        --------
+        The selection parser understands the following *keywords*:
+
+        Simple field selections
+        -----------------------
+        The following selections allow implicit OR of any number of values after
+        their keyword. For example::
+            'name HT1 HT2' == 'name HT1 or name HT2'
+        They also allow the use of the wildcard character '*', for example::
+            'name HT*'
+        Selects all atoms whose name starts with 'HT'
+
+        name *atom-name*
+            select by atom name (as given in the topology). Often, this is
+            force field dependent.
+            e.g. ``name CA`` (for C alpha atoms)
+            or ``name OW`` (for SPC water oxygen)
+        type *atom-type*
+            select by atom type; this is either a string or a number and
+            depends on the force field; it is read from the topology file
+            (e.g. the CHARMM PSF file contains numeric atom types).
+        resname *residue-name*
+            select by residue name, e.g. ``resname LYS``
+        segid *seg-name*
+            select by segid (as given in the topology),
+            e.g. ``segid 4AKE`` or ``segid DMPC``
+        altloc *alternative-location*
```

Comment:
I just realized how the new topology system might make documenting the selection language right here difficult. Completely fine for this PR, but we'll have to think about this. I imagine it might be as simple as saying "the token is the singular attribute name, followed by values to filter on."